frontend/features/page/sections/ContactsSection/ContactsSection.tsx

```diff
@@ -0,0 +1,675 @@
+  name: string;
+  value: string;
+}
+
+interface SuccessAction {
+  type: FormActionType.Success;
+}
+
+interface FailAction {
+  type: FormActionType.Failed;
+}
+
+interface SubmitAction {
+  type: FormActionType.Submit;
+}
+
+enum FormActionType {
+  UpdateField = "update-field",
+  Success = "success",
+  Failed = "failed",
+  Submit = "submit",
+}
+
+type FormAction = UpdateFieldAction | SuccessAction | FailAction | SubmitAction;
+
+interface FormState {
+  status: FormStatus;
+  values: {
+    name: string;
+    email: string;
+    message: string;
+  };
+  validationErrors: {
+    name: Nullable<string>;
+    email: Nullable<string>;
+    message: Nullable<string>;
+  };
+  submitError: string;
+}
+
+function reducer(state: FormState, action: FormAction) {
+  const regex = /^[^@\s]+@[^@\s\.]+\.[^@\.\s]+$/;
+  switch (action.type) {
+    case FormActionType.UpdateField: {
+      return {
+        ...state,
+        values: {
+          ...state.values,
+          [action.name]: action.value,
+        },
+        validationErrors: {
+          ...state.validationErrors,
+          [action.name]: "",
+        },
+        submitError: "",
+      };
+    }
+
+    case FormActionType.Submit: {
+      //Check errors
+
+      let error = false;
+      if (
+        state.values.name == "" ||
+        !state.values ||
+        state.values.name?.replace(/\s/g, "").length == 0
+      ) {
+        state.validationErrors.name = "Please insert your name";
+        error = true;
+      } else {
+        state.validationErrors.name = "";
+        if (
+          state.values.email == "" ||
+          !state.values.email ||
+          state.values.email?.replace(/\s/g, "").length == 0
+        ) {
+          state.validationErrors.email = "Please insert your email";
+          error = true;
+        } else {
+          if (!regex.test(state.values.email)) {
+            state.validationErrors.email = "Please insert a valid email";
+            error = true;
+          } else {
+            state.validationErrors.email = "";
+            if (state.values.message == "" || !state.values.message) {
+              state.validationErrors.message = "Please insert a message";
+              error = true;
+            } else {
+              state.validationErrors.message = "";
+            }
+          }
+        }
+      }
+
+      event?.preventDefault();
+
+      if (error) {
+        return {
+          ...state,
+          status: FormStatus.Idle,
+        };
+      }
+
+      return {
+        ...state,
+        status: FormStatus.Submitting,
+      };
+    }
+
+    case FormActionType.Success:
+      return {
+        // Display thanks page
+        ...state,
+        status: FormStatus.Submitted,
+      };
+
+    case FormActionType.Failed:
+      return {
+        // Resta in idle, setta errore e resetta i campi?
+        ...state,
+        values: {
+          name: "",
+          email: "",
+          message: "",
+        },
```

Comment:
I think we don't want this.. Say a network error occurs, as a use would you like to see all the inputs be cleared? Maybe you wrote a really long message, and now you lost it just because a network error occurred. Would you be happy with it? The user should be able to try again without efforts